superset-frontend/src/explore/controlPanels/TimePivot.js

```diff
@@ -17,47 +17,99 @@
  * under the License.
  */
 import { t } from '@superset-ui/translation';
-import { D3_TIME_FORMAT_OPTIONS } from '../controls';
+import { D3_FORMAT_OPTIONS } from '../controls';
+import {
+  lineInterpolation,
+  showLegend,
+  xAxisLabel,
+  bottomMargin,
+  xAxisFormat,
+  yLogScale,
+  yAxisBounds,
+  xAxisShowMinmax,
+  yAxisShowMinmax,
+  yAxisLabel,
+  leftMargin,
+} from './Shared_NVD3';
 
 export default {
   requiresTime: true,
   controlPanelSections: [
     {
       label: t('Query'),
       expanded: true,
-      controlSetRows: [['metric'], ['adhoc_filters'], ['freq']],
+      controlSetRows: [
+        ['metric'],
+        ['adhoc_filters'],
+        [
+          {
+            name: 'freq',
+            config: {
+              type: 'SelectControl',
+              label: t('Frequency'),
+              default: 'W-MON',
+              freeForm: true,
+              clearable: false,
+              choices: [
+                ['AS', 'Year (freq=AS)'],
+                ['52W-MON', '52 weeks starting Monday (freq=52W-MON)'],
+                ['W-SUN', '1 week starting Sunday (freq=W-SUN)'],
+                ['W-MON', '1 week starting Monday (freq=W-MON)'],
+                ['D', 'Day (freq=D)'],
+                ['4W-MON', '4 weeks (freq=4W-MON)'],
+              ],
+              description: t(
+                `The periodicity over which to pivot time. Users can provide
+            "Pandas" offset alias.
+            Click on the info bubble for more details on accepted "freq" expressions.`,
+              ),
+              tooltipOnClick: () => {
+                window.open(
+                  'https://pandas.pydata.org/pandas-docs/stable/timeseries.html#offset-aliases',
+                );
+              },
+            },
+          },
+        ],
+      ],
     },
     {
       label: t('Chart Options'),
       expanded: true,
       controlSetRows: [
-        ['show_legend', 'line_interpolation'],
+        [showLegend, lineInterpolation],
         ['color_picker', null],
       ],
     },
     {
       label: t('X Axis'),
       expanded: true,
       controlSetRows: [
-        ['x_axis_label', 'bottom_margin'],
-        ['x_axis_showminmax', 'x_axis_format'],
+        [xAxisLabel, bottomMargin],
+        [
+          xAxisShowMinmax,
+          {
```

Comment:
Let's create a util function for convenience and to avoid mistake in the future. 
Can also leverage typescript.

```ts
// Can add stronger typing per control type later
interface ControlConfig {
  type: string;
  label: string;
  description?: string;
  renderTrigger: boolean;
  [key: string]: unknown;
}

interface FieldConfig {
  name: string;
  config: ControlConfig;
}

export default function overrideFieldConfig(
  base: FieldConfig,
  override: Partial<ControlConfig>,
): FieldConfig {
  const { name, config } = base;
  return {
    name,
    config: { ...config, override },
  };
}
```

Reply:
Honestly, I just made the mistake because I was having a copy-paste party late at night. I don't think I would have made a mistake if I were actually trying to override a control with a specific intent.

My concern is that a function like this adds another onion layer for devs to figure out... in the end, having to figure out that you pass in a complete config and a partial config to make it work seems just as error-prone.

I think the spread syntax is straightforward enough... or we could re-enable people to use Object.assign if that's any more straightforward.

Reply:
Good news is, it seems like we're down to this, and an easy fast-follow PR. Excited this one is almost behind us, and appreciative of your time/effort on this.

Reply:
I agree for any utility functions we introduce we should also consider whether it makes sense to expose to future plugin developers. Ideally there should be a straightforward public API with which developers can do most of what they want without the need of diving into the internals of the plugin system.

Reply:
I don't feel that strongly about this one. Can drop.
My main intention was to start introducing type checking for the config. 
Perhaps can revisit typing later.